bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsDeviceHandler.java

```diff
@@ -49,12 +53,32 @@ public void initialize() {
         deviceId = config.deviceId;
 
         ElroConnectsBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler == null) {
+            // Thing status has already been updated in getBridgeHandler()
+            return;
+        }
 
-        if (bridgeHandler != null) {
-            bridgeHandler.setDeviceHandler(deviceId, this);
-            updateProperties(bridgeHandler);
-            updateDeviceName(bridgeHandler);
-            refreshChannels(bridgeHandler);
+        if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
+            ElroConnectsDevice device = bridgeHandler.getDevice(deviceId);
+            if (device != null) {
+                ElroDeviceType deviceType = TYPE_MAP.get(device.getDeviceType());
+                if ((deviceType == null) || !thing.getThingTypeUID().equals(THING_TYPE_MAP.get(deviceType))) {
+                    String msg = String.format("@text/offline.invalid-device-type [ \"%s\" ]", device.getDeviceType());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+                } else {
+                    updateStatus(ThingStatus.UNKNOWN);
```

Comment:
In case you are sure that `refreshChannels(bridgeHandler)` will set a thing status quickly and in the current thread, I do not see the need to set the thing status to UNKNOWN. It would lead to the sequence OFFLINE => UNKNOWN => ONLINE maybe for no reason.
I let you confirm as it simply depends on `refreshChannels` implementation (I have not checkoed).

Reply:
It should, but I found one case where it might fall through when an unexpected message was received. So I fixed that.